File: card-api/src/main/java/com/jean202/cardmizer/api/demo/DemoRecommendationRequest.java
package com.jean202.cardmizer.api.demo;

import java.util.Objects;
import java.util.Set;

public record DemoRecommendationRequest(
        String spendingMonth,
        long amount,
        String merchantName,
        String merchantCategory,
        Set<String> paymentTags
) {
    public DemoRecommendationRequest {
        if (spendingMonth == null || spendingMonth.isBlank()) {
            throw new IllegalArgumentException("spendingMonth must not be blank");
        }
        if (merchantName == null || merchantName.isBlank()) {
            throw new IllegalArgumentException("merchantName must not be blank");
        }
        paymentTags = Set.copyOf(Objects.requireNonNull(paymentTags, "paymentTags must not be null"));
    }
}

File: card-api/src/main/kotlin/com/jean202/cardmizer/api/demo/DemoRecommendationRequest.kt
package com.jean202.cardmizer.api.demo

class DemoRecommendationRequest(
    val spendingMonth: String,
    val amount: Long,
    val merchantName: String,
    val merchantCategory: String?,
    paymentTags: Set<String>,
) {
    val paymentTags: Set<String>

    init {
        require(spendingMonth.isNotBlank()) { "spendingMonth must not be blank" }
        require(merchantName.isNotBlank()) { "merchantName must not be blank" }
        this.paymentTags = paymentTags.toSet()
    }
}
